File: MoneyAI/ParentChildMatchers/AmazonOrderMatcher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CommonUtils;

namespace MoneyAI.ParentChildMatchers
{
    public class AmazonOrderMatcher : GenericOrderMatcher
    {
        public AmazonOrderMatcher(AccountInfo accountInfo)
            :base(accountInfo, @"shipping charge", @"tax charged", @"total promotions")
        {
        }
    }
}

File: packages/core/src/matching/amazon-order-matcher.ts
/**
 * Amazon order-history parent-child matcher.
 *
 * Ported from C# `AmazonOrderMatcher`. A thin subclass of
 * {@link GenericOrderMatcher} that supplies Amazon-specific
 * provider attribute names for shipping, tax, and discount.
 *
 * @module
 */

import type { AccountInfo } from '../models/account-info.js';
import { GenericOrderMatcher } from './generic-order-matcher.js';

export class AmazonOrderMatcher extends GenericOrderMatcher {
    constructor(accountInfo: AccountInfo) {
        super(accountInfo, 'shipping charge', 'tax charged', 'total promotions');
    }
}
